lib/perl/Genome/InstrumentData/AlignmentResult/Merged.pm

```diff
@@ -707,4 +786,54 @@ sub scalar_property_from_underlying_alignment_results {
 }
 
 
+sub _lock_per_lane_alignment {
+    my $self = shift;
+
+    for my $alignment ($self->collect_individual_alignments) {
+        unless ($alignment->get_merged_alignment_results) {
+            my @bams = $alignment->alignment_bam_file_paths;
+            unless (@bams) {
+               die $self->error_message("Alignment result with class (%s) and id (%s) has neither ".
+                   "merged results nor valid bam paths. This likely means that this alignment result ".
+                   "needs to be removed and realigned because data has been lost. ".
+                   "Please create an apipe-support ticket for this.", $alignment->class, $alignment->id);
+               #There is no way to recreate per lane bam if merged bam
+               #does not exist and per lane bam is removed. Software
+               #result of this per lane alignment needs to be removed and
+               #this per lane instrument data needs to be realigned
+               #with that aligner.
+            }
+
+            unless ($ENV{UR_DBI_NO_COMMIT}) {
+                my $lock_var = File::Spec->join($ENV{GENOME_LOCK_DIR}, 'genome', __PACKAGE__, 'lock-per-lane-alignment-'.$alignment->id);
```

Comment:
Just trying to think about your locking problem - are you guaranteed that all of the alignment objects returned by collect_individual_alignments will be unique?